lib/tapioca/dsl/compilers/job_iteration.rb

```diff
@@ -0,0 +1,120 @@
+# typed: strict
+# frozen_string_literal: true
+
+return unless defined?(JobIteration::Iteration)
+
+module Tapioca
+  module Dsl
+    module Compilers
+      class JobIteration < Compiler
+        extend T::Sig
+
+        ConstantType = type_member { { fixed: T.class_of(::JobIteration::Iteration) } }
+
+        sig { override.void }
+        def decorate
+          return unless constant.instance_methods(false).include?(:build_enumerator)
+
+          root.create_path(constant) do |job|
+            method = constant.instance_method(:build_enumerator)
+            constant_name = name_of(constant)
+            expanded_parameters = compile_method_parameters_to_rbi(method).reject do |typed_param|
+              typed_param.param.name == "cursor"
+            end
+
+            job.create_method(
+              "perform_later",
+              parameters: perform_later_parameters(expanded_parameters, constant_name),
+              return_type: "T.any(#{constant_name}, FalseClass)",
+              class_method: true,
+            )
+
+            job.create_method(
+              "perform_now",
+              parameters: expanded_parameters,
+              return_type: "void",
+              class_method: true,
+            )
+
+            job.create_method(
+              "perform",
+              parameters: expanded_parameters,
+              return_type: "void",
+              class_method: false,
+            )
+          end
+        end
+
+        private
+
+        sig do
+          params(
+            parameters: T::Array[RBI::TypedParam],
+            constant_name: T.nilable(String),
+          ).returns(T::Array[RBI::TypedParam])
+        end
+        def perform_later_parameters(parameters, constant_name)
+          if ::Gem::Requirement.new(">= 7.0").satisfied_by?(::ActiveJob.gem_version)
+            parameters.reject! { |typed_param| RBI::BlockParam === typed_param.param }
+            parameters + [create_block_param(
+              "block",
+              type: "T.nilable(T.proc.params(job: #{constant_name}).void)",
+            )]
+          else
+            parameters
+          end
+        end
+
+        def compile_method_parameters_to_rbi(method_def)
```

Comment:
Do you need to copy this complex method over? I would have instead done a call to `compile_method_parameters_to_rbi` and then walked over the result to find all required params of type `T::Types:FixedHash` and replace them with the keyword parameters instead.

Reply:
Ah, sorry, those are also strings. I guess you could do:
```ruby
signature = signature_of(method_def)
fixed_hash_args = signature.arg_types.select { |arg_type| arg_type[1].is_?(T::Types::FixedHash) }
```
and loop over those to match the params to the ones you need to replace.